apps/hash-frontend/src/components/hooks/block-protocol-functions/ontology/use-block-protocol-query-property-types.ts

```diff
@@ -45,6 +45,7 @@ export const useBlockProtocolQueryPropertyTypes = (): {
           constrainsValuesOn: { outgoing: 255 },
           constrainsPropertiesOn: { outgoing: 255 },
           ...graphResolveDepths,
+          includeArchived: true,
```

Comment:
I think we want to make this an argument passed to the hook, rather than true – at the moment, archived property types will show up as options in the dropdowns for the entity type selector, which I don't think we want (otherwise there's no real point in archiving them). This hook can be called with `includeArchived: true` from the types table page.

Reply:
Currently this context is global, so this hook is being used in a single place to populate that context, where we now want to also get the archived property types.

But I appreciate your point regarding the property type selector now displaying archived property types, so I've updated the `useLatestPropertyTypesContextRequired` hook to accept an `includeArchived` parameter, which by default filters any archived entities from the response, in https://github.com/hashintel/hash/pull/3193/commits/fdfacaa9d8e64d36134cffdc1c837c634096e1f0. This should prevent archived property types from being included in things by default, and allow the property types table to continue accessing the archived property types via `includeArchived: true`.

I've also removed a `useLatestPropertyTypes` hook which was used previously, which didn't require the context to be set. However this didn't make much sense to me, as the context provider is located in `_app.page.tsx` so no matter where you require the context in the app it should be available. I've replaced all usages of `useLatestPropertyTypes` with `useLatestPropertyTypesContextRequired`.

Reply:
I would just call the hook `useLatestPropertyTypes`, we can add `useLatestPropertyTypesOptional` if we ever need it – I think `useLatestPropertyTypesContextRequired` is too verbose. I appreciate we have `useEntityTypesContextRequired` elsewhere but I don't think it's necessary, my default expectation would be that a hook relying on context would crash if the context wasn't available.

---

> Currently this context is global, so this hook is being used in a single place to populate that context, where we now want to also get the archived property types.

I don't think we need to fix this now but we will want to change this at some point.

The drawback of this is that now the global context for property types which loads on every page is fetching more types than it needs to, and if any else uses this hook they are also getting archived types by default. I think it would be better if the types page did its own querying – it doesn't need to go via the `useBlockProtocol` hook either, which adds a level of indirection around just using Apollo Client directly, e.g. 

```ts
  const { data, loading } = useQuery<
    QueryPropertyTypesQuery,
    QueryPropertyTypesQueryVariables
  >(queryPropertyTypesQuery, {
    fetchPolicy: "cache-and-network",
    variables: {
      ...zeroedGraphResolveDepths,
      constrainsValuesOn: { outgoing: 255 },
      constrainsPropertiesOn: { outgoing: 255 },
      includeArchived: true,
    },
  });

  const latestPropertyTypes = useMemo(
    () =>
      data
        ? Object.fromEntries(
            getRoots<PropertyTypeRootType>(
              data.queryPropertyTypes as Subgraph<PropertyTypeRootType>,
            ).map((propertyType) => {
              return [propertyType.schema.$id, propertyType];
            }),
          )
        : null,
    [data],
  );
```

This would have the advantage of also being able to vary `includeArchived` based on user choice on the type page, rather than filtering client-side, which we'll also want to do with other filters and pagination on this page once we need to worry about there being lots of data.

By using a `cache-and-network` policy we ensure that any new property types added during the session would be fetched when visiting the page again, so we don't have to worry about refetching if we change things elsewhere (unless we happen to be on this page somehow, which isn't possible at the moment).

Reply:
Agreed, have renamed the hook to `useLatestPropertyTypes`, and have also added the `includeArchived` prop to  preparation for moving the context out of `_app.tsx` have added the `includeArchived` prop to the `LatestPropertyTypesContextProvider` component incase we want to move it out of `_app.page.tsx`, and not include archived property types, as part of https://github.com/hashintel/hash/pull/3193/commits/674ad4cb0fcc1ba69a4b5c5cdbdeead2ecf1a5f9